### PR TITLE
Add Bolometric lightcurve calculation to mcmc case

### PR DESCRIPTION
**Pull request by Tomariebenami**
Add a Bolometric Calculation to the MCMC case of `calculate_bolometric()`.

This fix calculates the bolometric luminoscity from the Stefan Boltzmann equation, for the mcmc case fit of the planck equation. 

This Fix includes:
1. Fixing `stefan_boltzmann()` equation - it is not able to also accept only T and R, without respective errors. This is used to calculate the luminosity for the individual samples. 
2. Changing the pseudo bolometric luminosity row names from `L_mcmc, dL_mcmc0, dLmcmc1` to `L_mcmc_opt, dL_mcmc0_opt, dLmcmc1_opt`. This now follows the nomenclature of the scipy case, where the pseudo luminosity takes the name `L_opt`.
3. Add in `calculate_bolometric()` the calculation of the bolometric luminosity. The row names `L_mcmc, dL_mcmc0, dLmcmc1` nor correspon to the full bolometric luminosity.

### CODE REVIEW

File: lightcurve_fitting/bolometric.py [hunk 1]
```diff
@@ -728,8 +735,9 @@ def calculate_bolometric(lc, z=0., outpath='.', res=1., nwalkers=10, burnin_step
         color_mags, color_dmags, color_lolims, color_uplims = calc_colors(epoch1, colors)
 
         row = [mjdavg, dmjd0, dmjd1,
-               temp, radius, dtemp, drad, lum, dlum, L_opt,
-               T_mcmc, R_mcmc, dT0_mcmc, dT1_mcmc, dR0_mcmc, dR1_mcmc, L_mcmc, dL_mcmc0, dL_mcmc1,
+               temp, radius, dtemp, drad, lum, dlum, L_opt, L_mcmc, dL_mcmc0, dL_mcmc1,
+               T_mcmc, R_mcmc, dT0_mcmc, dT1_mcmc, dR0_mcmc, dR1_mcmc, 
+               L_mcmc_opt, dL_mcmc0_opt, dL_mcmc1_opt,
                L_int, nfilt] + color_mags + color_dmags
```
Comment (griffin-h) [inline lightcurve_fitting/bolometric.py]: If you're adding new columns to the output row, you'll have to add them to the table initialization in lines 641-654 as well

File: lightcurve_fitting/bolometric.py [hunk 2]
```diff
@@ -350,35 +350,38 @@ def group_by_epoch(lc, res=1., also_group_by=()):
...
     """
-    lum = 4 * np.pi * radius ** 2 * sigma_sb * temp ** 4
-    dlum = 8 * np.pi * sigma_sb * (radius ** 2 * temp ** 8 * drad ** 2
+    if dtemp == None and drad == None:
+        return 4 * np.pi * radius ** 2 * sigma_sb * temp ** 4
+    else:
+        lum = 4 * np.pi * radius ** 2 * sigma_sb * temp ** 4
+        dlum = 8 * np.pi * sigma_sb * (radius ** 2 * temp ** 8 * drad ** 2
                                    + 4 * radius ** 4 * temp ** 6 * dtemp ** 2
                                    + 4 * radius ** 3 * temp ** 7 * covTR) ** 0.5
```
Comment (griffin-h) [inline lightcurve_fitting/bolometric.py]: To make sure this never crashes, just add this between the original lines 377 and 378:
```
if dtemp is None or drad is None or covTR is None:
    return lum
```

**Review by griffin-h — CHANGES_REQUESTED** [review]
The calculation looks good! I'm a little hesitant to change the meaning of existing variables though. This will really confuse people using old versions of the code. (I should have chosen better names to start with, but too late for that!) Can I suggest this naming scheme:

* All luminosities start with capital `L`. The uncertainties start with `dL`.
* If they are from the least squares method, use no suffix. If they are from the MCMC method, use the suffix `_mcmc`.
* If they are pseudobolometric, use no suffix. If they are true bolometric, use the suffix `_bol`.

So then we get: `L_bol` (previously `lum`), `L` (previously `L_opt`), `L_mcmc` (unchanged), and `L_mcmc_bol` (new), plus their associated uncertainties. Ideally, we would also leave `lum` and `L_opt` in the table for backwards compatibility (for now).

Once you do this, can you also update the documentation to explain this convention, and update the release history to log your changes?

**Comment by Tomariebenami** [comment]
I agree with you on the variable names. Your suggestion is indeed better, I'll implement it in.

About the documentation and release history - Of Course! I'll add that in too.

**Comment by Tomariebenami** [comment]
OK, 

I've updated the code into the new naming convention. (Old luminosity convention works still)
A short warning messege was added to make sure people notice the difference.
Documentation has been added. Release history info added.

**Review by griffin-h — APPROVED** [review]
Thanks for all this! I made a few more changes to improve backward compatibility and added a longer description of the changes. Let me know if everything still looks right to you, and I'll merge it to the dev branch.

**Comment by Tomariebenami** [comment]
Looks great overall, I went over it now and did not find anything I think needs changing. 

Hope it helped!

**Pull request merged**